src/utils/workflow-utils.ts

```diff
@@ -1,8 +1,58 @@
-import { generateCaptureDraftSpec, modifyDraftSpec } from 'api/draftSpecs';
+import {
+    DraftSpecsExtQuery_ByCatalogName,
+    modifyDraftSpec,
+} from 'api/draftSpecs';
 import { DraftSpecQuery } from 'hooks/useDraftSpecs';
+import { isEmpty } from 'lodash';
 import { CallSupabaseResponse } from 'services/supabase';
 import { ResourceConfigDictionary } from 'stores/ResourceConfig/types';
-import { Schema } from 'types';
+import { EntityWithCreateWorkflow, Schema } from 'types';
+import { ConnectorConfig } from '../../flow_deps/flow';
+
+// TODO (typing): Narrow the return type for this function.
+export const generateTaskSpec = (
+    entityType: EntityWithCreateWorkflow,
+    connectorConfig: ConnectorConfig,
+    resourceConfigs: ResourceConfigDictionary | null,
+    existingTaskData: DraftSpecsExtQuery_ByCatalogName | null
+) => {
+    const draftSpec = isEmpty(existingTaskData)
+        ? {
+              bindings: [],
+              endpoint: {},
+          }
+        : existingTaskData.spec;
+
+    draftSpec.endpoint.connector = connectorConfig;
+
+    if (resourceConfigs) {
+        const collectionNameProp =
+            entityType === 'capture' ? 'target' : 'source';
+
+        Object.keys(resourceConfigs).forEach((collectionName) => {
+            const resourceConfig = resourceConfigs[collectionName].data;
+
+            const existingBindingIndex = draftSpec.bindings.findIndex(
+                (binding: any) => binding[collectionNameProp] === collectionName
+            );
+
+            if (existingBindingIndex > -1) {
+                draftSpec.bindings[existingBindingIndex].resource = {
+                    ...resourceConfig,
+                };
```

Comment:
Why are we using the spread operator here / down below?

Reply:
Ahhh... I think I made that change and can not longer remember what was driving it :(